client/json-rpc/src/client.rs

```diff
@@ -148,22 +148,7 @@ pub enum JsonRpcAsyncClientError {
 impl JsonRpcAsyncClientError {
     pub fn is_retriable(&self) -> bool {
         match self {
-            JsonRpcAsyncClientError::ClientError(e) => {
-                // TODO: Add following condition when is_request is added to reqwest stable release
-                // // Error sending request
-                //  if e.is_request() {
-                //      return true;
-                //  }
-                if e.is_timeout() {
-                    return true;
-                }
-
-                if let Some(status) = e.status() {
-                    // Returned status code indicates a server error
-                    return status.is_server_error();
-                }
-                false
-            }
+            JsonRpcAsyncClientError::ClientError(_) => true,
```

Comment:
I am not sure this is necessarily true. Are we 100% sure we'd not receive `ClientError` in case there is a problem with the request itself?